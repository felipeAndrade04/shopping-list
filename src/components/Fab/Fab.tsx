import React from 'react';
import { Feather } from '@expo/vector-icons';

import * as S from './Fab.styles';
import { colors } from '@app/theme';

export function Fab() {
  return (
    <S.Container activeOpacity={0.7}>
      <Feather name="plus" color={colors.white} size={32} />
    </S.Container>
  );
}
