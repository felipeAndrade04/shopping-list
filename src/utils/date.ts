const months = ['jan', 'fev', 'mar', 'abr', 'mai', 'jun', 'jul', 'ago', 'set', 'out', 'nov', 'dez'];

export function formatDate(date: Date) {
  const day = date.getDay();
  const month = date.getMonth();

  return `${day} de ${months[month]}`;
}
