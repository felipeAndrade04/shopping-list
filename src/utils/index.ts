export * from './delay';
export * from './date';
