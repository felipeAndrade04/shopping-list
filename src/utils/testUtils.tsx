import React from 'react';
import { authReducer } from '@app/store';
import { ReactNode } from 'react';
import { Provider } from 'react-redux';
import { ThemeProvider } from 'styled-components/native';
import * as theme from '@app/theme';
import { configureStore } from '@reduxjs/toolkit';

let store = configureStore({
  reducer: {
    auth: authReducer,
  },
});

export const wrapper = ({ children }: { children: ReactNode }) => (
  <ThemeProvider theme={theme}>
    <Provider store={store}>{children}</Provider>
  </ThemeProvider>
);

export function resetStore() {
  store = configureStore({
    reducer: {
      auth: authReducer,
    },
  });
}
