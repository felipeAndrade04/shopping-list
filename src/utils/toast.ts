import EventManager from './EvenManager';

export type ToastType = 'success' | 'info' | 'error';

export interface Toast {
  id?: string;
  type: ToastType;
  text: string;
  duration?: number;
}

export const toastEventManager = new EventManager();

export default function toast({ type, text, duration }: Toast) {
  toastEventManager.emit<Toast>('addtoast', {
    id: new Date().getTime().toString(),
    type,
    text,
    duration,
  });
}
