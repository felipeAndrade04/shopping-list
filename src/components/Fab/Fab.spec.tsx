import React from 'react';
import { render } from '@testing-library/react-native';
import { Fab } from './Fab';
import { wrapper } from '@app/utils/testUtils';

describe('<Fab />', () => {
  it('Should render correctly', () => {
    const tree = render(<Fab />, { wrapper }).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
