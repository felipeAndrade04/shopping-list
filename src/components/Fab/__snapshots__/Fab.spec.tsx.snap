// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Fab /> Should render correctly 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#009Cf7",
      "borderRadius": 999,
      "height": 60,
      "justifyContent": "center",
      "opacity": 1,
      "width": 60,
    }
  }
>
  <Text />
</View>
`;
